### DISCUSSION TITLE
Display non-editable fields normally

### DISCUSSION THREAD

**Discussion by JoshTheDerf** [formfcw/directus-extension-editable-layout#2] [Feature Request]
We'd love to adopt this extension for some use-cases, especially some of our upcoming plans where inline editing will speed things up a great deal.

While testing this extension, we noticed that only fields that can be edited are allowed to be displayed in the table. Relational/Nested fields are not displayed. 

We have a fairly complex data model that has multiple layers of nesting for various purposes. Even though we might not need to edit nested information, we do need to be able to see that nested information in the table layout.

Would it be possible and reasonable to:
- Allow adding all fields, including relational/nested ones that are currently supported by the table layout.
- Use native displays for fields not currently supported for editing. 
- Skip over fields that are not currently supported for editing when navigating around the layout with the keyboard.

**Comment by formfcw** [comment]
Thanks for your request!

Basically, it was always my intention to only make fields available that could also be edited in the item/detail view. Therefore, you will find the restrictions in the code below for fields that are `is_primary_key`, `hidden`, or `readonly`.

https://github.com/formfcw/directus-extension-editable-layout/blob/6836ce0bb174aa5c217177535a089f8c934b7447/src/index.ts#L554-L562

_Off topic: The reason why Flexible Editor is not allowed is that it depends on a separate M2A field which makes editing here quite complex at the moment. And the reason why relational fields are not allowed at the moment is that they have a problem with the display at the moment._

I think it is no big deal to allow `is_primary_key`/`hidden`/`readonly` fields and display them as read-only. So you need to implement a condition to determine if it is editable or not. If it is editable, render the `<editable-cell />` component, otherwise just render `<render-display />` — therefore, you can copy lines 99–103.

https://github.com/formfcw/directus-extension-editable-layout/blob/6836ce0bb174aa5c217177535a089f8c934b7447/src/layout.vue#L94-L113

In the `Add field` dropdown it would be nice if you could indicate, that these fields are read-only by simply appending `(Readonly)` to that field name — therefore use the `$t:readonly` translation.

I think that should do it. I am really happy to see a PR for it.

For now, I would recommend keeping relation fields excluded. If you still want to work on it, please do so in a separate PR.
